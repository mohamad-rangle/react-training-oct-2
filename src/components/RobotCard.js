import React from 'react';
import PropTypes from 'prop-types';
import { Avatar } from './Avatar';
import { Heading } from './Heading';
import { Text } from './Text';
import { TiltCard } from './TiltCard';

export const RobotCard = ({ uid, name, email, className, ...props }) => (
  <TiltCard className={`pa3 bg-near-white br2 tc ${className}`} {...props}>
    <Avatar uid={uid} />
    <Heading level={2} size={4}>
      {name}
    </Heading>
    <Text color="green">{email}</Text>
  </TiltCard>
);

RobotCard.PropTypes = {
  uid: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  email: PropTypes.string.isRequired,
};
