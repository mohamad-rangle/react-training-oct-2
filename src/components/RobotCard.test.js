import React from 'react';
import renderer from 'react-test-renderer';
import { RobotCard } from './RobotCard';

test('RobotCard', () => {
  const tree = renderer
    .create(<RobotCard uid={1} name="Alice X" email="alice@test.com" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
