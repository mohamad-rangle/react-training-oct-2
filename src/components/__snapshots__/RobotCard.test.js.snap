// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RobotCard 1`] = `
<div
  className="pa3 bg-near-white br2 tc "
>
  <img
    alt="user profile"
    className="br-100 w4 h4 "
    src="https://robohash.org/1"
  />
  <h2
    className="sans-serif dark-gray f4"
  >
    Alice X
  </h2>
  <p
    className="green f5 sans-serif"
  >
    alice@test.com
  </p>
</div>
`;
